### ISSUE TITLE
lint: disable-next-line on max-params doesn't work when placed before function declaration

### ISSUE THREAD

**Issue by paul-sachs** [oxc-project/oxc#10115]
The disable-next-line directive doesn't seem to span large enough to cover the full function declaration. The following code snippets should not trigger lint errors:

```tsx
// eslint-disable-next-line max-params
function someMaxParamsFunction(
  param1: number,
  param2: number,
  param3: number,
  param4: number,
  param5: number,
  param6: number
) {

}
```

A workaround is to put the disable-next-line directive before the list of params instead but this doesn't read as nicely and is at odds with how eslint handles this exact scenario.

```ts
function someMaxParamsFunction(
  // eslint-disable-next-line max-params
  param1: number,
  param2: number,
  param3: number,
  param4: number,
  param5: number,
  param6: number
) {

}
```

**Comment by ulrichstark** [comment]
Hey @paul-sachs, thanks for reporting! Your issue should be fixed by #10141.

**Comment by paul-sachs** [comment]
Fantastic! I was just looking at that code but I'm still very new to rust so was slow going at trying to figure out what exactly needed fixing. Looking forward to the merge.

**Comment by ulrichstark** [comment]
Your issue is fixed in version 0.16.4.
